eclass/acct-user.eclass

```diff
@@ -222,22 +222,30 @@ elockuser() {
 	eislocked "$1"
 	[[ $? -eq 0 ]] && return 0
 
+	local opts
+	[[ -n ${ROOT} ]] && opts=( -R "${ROOT}" )
```

Comment:
According to shadow code: https://github.com/shadow-maint/shadow/blob/master/libmisc/root_flag.c#L92
`-R` only drop privilege / chdir / chroot, so if the sandbox is bypassed, it was already the case.

Reply:
I meant that chroot'ing executes code from the $ROOT which you might not expect. Plus this is in an unsandboxed function (since {pre,post}{inst,rm} can change / even if $ROOT != /)

Also, if we're gonna chroot using individual tools we might as well run all of {pre,post}{inst,rm} in a chroot and this would fix all of these kinds of issues (and make these kind of changes unnecessary)

I'm not a gentoo-dev though so idk what's the concensus on this

We use --prefix in the patches from the bugtracker

Reply:
Agreed on the fact that all of this depend of what is installed on $ROOT (see the case separation with BSD)

The chroot idea is not bad but is limited to a set of compatible architecture between the host and the target (AMD64 -> X86 / AMD64 for example).
If you want to compile another architecture which is not natively compatible with your host, like AARCH64 for AMD64 host, you will have to use a Qemu/KVM trick which can works but, on my side, can have some heavy performance issue (+ you need to have a stage3 already decompressed).

Reply:
Yeah that's fair, realized that after posting. But that applies to shadow too, have you tested this with non-compatible targets? usermod -R is basically a glorified chroot so I don't think this is different

Reply:
Aka the difference between -R and --prefix seems to be NIS/LDAP/pam support, but those require the TARGET's binaries to be supported I assume (pam library/niscd/sssd)

Reply:
My test are actually done with: `aarch64-linux-gnu-emerge`

Reply:
I did further testing by decompressing a aarch64 stage3 and adding a user.
You're right, `-R` fails where `--prefix` works. After that, the home creation failed since chown failed to get the user (if it is not installed on the host).